tests/linen/summary_test.py

```diff
@@ -417,12 +417,11 @@ def __call__(self, x):
 
     lstm = LSTM(features=128)
 
-    with jax.check_tracer_leaks(True):
```

Comment:
I think what is going on is that because of the new abstract eval, some tracers are temporarily held as captures and this makes JAX panic so I had to remove it. I debugged it for a while bug couldn't figure it out. I should add a test using scan without `tabulate` to make sure that leakage is truly not happening.

Reply:
I looked into this again this `eval_shape` is making `jax.check_tracer_leaks` panic, something to do with the `Scope`, I might be doing this wrong.

https://github.com/google/flax/pull/3228/files#diff-1b115dbd8245bfadd0e87d27514d2b7e3f7aba95c6ddc4f4674f1bdb1c5fa4c5R918